packaging_automation/common_tool_methods.py

```diff
@@ -290,3 +323,154 @@ def remove_cloned_code(exec_path: str):
+    for key_id in keys:
+        delete_command = f"gpg --batch --yes --delete-key {key_id}" if key_type == GpgKeyType.public \
+            else f"gpg --batch --yes --delete-secret-key {key_id}"
+        output = run_with_output(delete_command)
+        if output.returncode == 0:
+            print(f"{key_type.name.capitalize()} key with the id {key_id} deleted")
+        #
+        elif output.returncode == 2:  # Key does not exist in keyring
+            continue
+        else:
+            print(f"Error {output.stderr.decode('ascii')}")
+            break
+
+
+def delete_public_gpg_key_by_name(name: str):
+    delete_gpg_key_by_name(name, GpgKeyType.public)
+
+
+def delete_private_gpg_key_by_name(name: str):
+    delete_gpg_key_by_name(name, GpgKeyType.private)
+
+
+def delete_all_gpg_keys_by_name(name: str):
+    delete_private_gpg_key_by_name(name)
+    delete_public_gpg_key_by_name(name)
+
+
+def get_secret_key_by_fingerprint_without_password(fingerprint: str) -> str:
+    gpg = gnupg.GPG()
+
+    private_key = gpg.export_keys(fingerprint, secret=True, expect_passphrase=False)
+    if private_key:
+        return private_key
+    else:
+        raise ValueError(
+            f"Error while getting key. Most probably packaging key is stored with password. "
+            f"Please check the password and try again")
+
+
+def get_secret_key_by_fingerprint_with_password(fingerprint: str, passphrase: str) -> str:
+    gpg = gnupg.GPG()
+
+    private_key = gpg.export_keys(fingerprint, secret=True, passphrase=passphrase)
+    if private_key:
+        return private_key
+    else:
+        raise ValueError(
+            f"Error while getting key. Most probably packaging key is stored with password. "
+            f"Please check the password and try again")
+
+
+def transform_key_into_base64_str(key: str) -> str:
+    # while signing packages base64 encoded string is required. So first we encode key with ascii and create a
+    # byte array than encode it with base64 and decode it with ascii to get the required output
+    return base64.b64encode(key.encode("ascii")).decode("ascii")
+
+
+def define_rpm_public_key_to_machine(fingerprint: str):
+    with open("rpm_public.key", "w") as writer:
+        subprocess.run(shlex.split(f"gpg --export -a {fingerprint}"), stdout=writer)
+    run("rpm --import rpm_public.key")
+    os.remove("rpm_public.key")
+
+
+def delete_rpm_key_by_name(summary: str):
+    rpm_keys = get_rpm_keys()
+    for key in rpm_keys:
+        if has_rpm_key_given_summary(key, summary):
+            run(f"rpm -e {key}")
+            print(f"RPM key with id {key} was deleted")
+
+
+def get_rpm_keys():
+    result = run_with_output("rpm -q gpg-pubkey")
+    if result.stderr:
+        raise ValueError(f"Error:{result.stderr.decode('ascii')}")
+    output = result.stdout.decode("ascii")
+    key_lines = output.splitlines()
+    return key_lines
+
+
+def has_rpm_key_given_summary(key: str, summary: str):
```

Comment:
Also, if we will keep this function (see the comment above), maybe `has_rpm_key_given_summary` -> `rpm_key_matches_summary`.